### PR TITLE
Calypso build: Require resolve babel presets and plugins

### PR DESCRIPTION
**Pull request by sirreal**
Observed in https://github.com/Automattic/jetpack/pull/13503 that when there are multiple versions of babel plugins and presets in a project, the default babel preset/plugin resolution fails.

Use `require.resolve` to ensure packages are found and resolved to the correct version.

## Testing
- Calypso builds
- Calypso monorepo apps build and work correctly
- Link and use for https://github.com/Automattic/jetpack/pull/13503

### CODE REVIEW

**Comment by jsnajdr** [comment]
How is `require.resolve` different from the resolution that Babel would otherwise do?

> when there are multiple versions of babel plugins and presets in a project

Does that mean there are different NPM package versions in different subdirectories of `node_modules`? Or something else?

Even after reading the Jetpack issue, I don't understand at all what this change does and what problem it addresses.

**Comment by sirreal** [comment]
> How is `require.resolve` different from the resolution that Babel would otherwise do?
> 
> > when there are multiple versions of babel plugins and presets in a project
> 
> Does that mean there are different NPM package versions in different subdirectories of `node_modules`? Or something else?

Sorry this was unclear. It appears that in other projects there may be different versions of the transitive dependencies or transitive dependencies that simply are not dedupe and lifted to the root for whatever reason. This results in the module being unreachable when babel is invoked in the project root ("normal" resolution"). However, if the module is resolved from these files, we should be guaranteed to resolve it correctly.

In Jetpack, this manifests with `@babel/plugin-proposal-class-properties` being located at`node_modules/@automattic/calypso-build/node_modules/@babel/plugin-proposal-class-properties`, which babel cannot find when invoked at the root.

[`@wordpress/babel-preset-default` follows the same pattern.](https://github.com/WordPress/gutenberg/blob/dd415505ad93fb42cd3dc2d746a2610795101ce1/packages/babel-preset-default/index.js#L58-L72)

**Comment by jsnajdr** [comment]
> In Jetpack, this manifests with @babel/plugin-proposal-class-properties being located atnode_modules/@automattic/calypso-build/node_modules/@babel/plugin-proposal-class-properties, which babel cannot find when invoked at the root.

Thanks, I think I understand it better now. Jetpack has `@babel/core` as a root (top-level) dependency. The Core code, when running, loads a Babel config file from `calypso-build`. That file references `@babel/plugin-proposal-class-properties`, and when trying to resolve this module from root, the Babel Core of course doesn't look for it in `node_modules/@automattic/calypso-build/node_modules`.

With `require.resolve`, the resolution is done in the context of the Babel config file, which is inside `calypso-build`, and the plugin is found.

File: webpack.config.js [hunk 1]
```diff
@@ -154,6 +154,12 @@ if ( isDevelopment || isDesktop ) {
 const cssFilename = cssNameFromFilename( outputFilename );
 const cssChunkFilename = cssNameFromFilename( outputChunkFilename );
 
+const fileLoader = FileConfig.loader( {
+	outputPath: path.join( __dirname, 'public', 'images' ),
+	publicPath: '../images/',
+	emitFile: browserslistEnv === 'evergreen', // Only output files once.
+} );
```
Comment (jsnajdr) [inline webpack.config.js]: This doesn't seem related?

Comment (sirreal) [reply webpack.config.js]: Indeed, that commit wasn't intended to be part of this PR. Rolled it back, thanks 🙇

**Comment by ockham** [comment]
I seem to have trouble testing this :confused: 

I tried `yarn link` from the `packages/calypso-build` directory of this branch, and then in my local Jetpack repo, I checked out `update/calypso-build-v4` (https://github.com/Automattic/jetpack/pull/13503) and did

```
yarn distclean && yarn && yarn link @automattic/calypso-build && yarn build
```

That worked, and made me curious if the ESM->CJS changes in https://github.com/Automattic/jetpack/pull/13503 were even needed, so I checked out Jetpack `master` and repeated that command. That also worked!

To make sure I wasn't getting false positives, I switched my Calypso repo back to `master`, did `npm run distclean and npm ci` from its root, and `yarn unlink && yarn link` from `packages/calypso-build`, and finally switched back to my Jetpack repo and ran the command above again. Once more, it built successfully -- with both repos on `master`, and Jetpack linking to Calypso's `@automattic/calypso-build`, i.e. without the changes from this PR  :confused: 

It seems like `yarn link` is skewing our results here :confused:

**Review by ockham — APPROVED** [review]
Tested this by just copy-pasting the modified files into my Jetpack repo's `node_modules/@automattic/calypso-build/` dir (while on Jetpack `master`, with `@automattic/calypso-buld` bumped to `4.0.1`), and it's working great :+1: 

Thanks a lot @sirreal! :shipit:
